prowler/providers/azure/azure_provider.py

```diff
@@ -244,14 +379,94 @@ def setup_session(
 
         return credentials
 
-    def __check_service_principal_creds_env_vars__(self):
+    @staticmethod
+    def test_connection(
+        az_cli_auth=False,
+        sp_env_auth=False,
+        browser_auth=False,
+        managed_identity_auth=False,
+        tenant_id=None,
+        region="AzureGlobal",
+        raise_exception=True,
+    ) -> bool:
```

Comment:
```suggestion
    ) -> Connection:
```

Please merge from `master` to get the `Connection` object.

Reply:
Done